examples/small-study/src/study.js

```diff
@@ -0,0 +1,144 @@
+// put the config in the scope so that background can see it.
+const STUDYCONFIG = (this.STUDYCONFIG = {
+  // activeExperimentsTag
+  activeExperimentName: "demoStudy",
+  // uses shield|pioneer pipeline, watches those permissions
+  pattern: "shield",
+  telemetry: {
+    send: true, // assumed false. Actually send pings?
+    removeTestingFlag: false, // Marks pings as testing, set true for actual release
+  },
+  endings: {
+    /** standard endings */
+    "user-disable": {
+      baseUrl: "http://www.example.com/?reason=user-disable",
+    },
+    ineligible: {
+      baseUrl: "http://www.example.com/?reason=ineligible",
+    },
+    expired: {
+      baseUrl: "http://www.example.com/?reason=expired",
+    },
+    dataPermissionsRevoked: {
+      baseUrl: null,
+      study_state: "ended-neutral",
+    },
+
+    /** User defined endings */
+    "some-study-defined-ending": {
+      study_state: "ended-neutral",
+      baseUrl: null,
+    },
+  },
+  logLevel: 10,
+  /* Study branches and sample weights, overweighing feature branches */
+  weightedVariations: [
+    {
+      name: "feature-active",
+      weight: 1.5,
+    },
+    {
+      name: "feature-passive",
+      weight: 1.5,
+    },
+    {
+      name: "control",
+      weight: 1,
+    },
+  ],
+  variation: "feature-active", // optional, overrides
+});
+
+// This is a study Study template.
+class BaseStudy {
+  async sendTelemetry(payload) {
+    const pingSchema = {
+      type: "object",
+      // all keys and values must be strings to be valid `shield-study-addon` pings
+      additionalProperties: {
+        type: "string",
+      },
+    };
+    const validation = browser.study.validateJSON(payload, pingSchema);
+    if (!validation.isValid) {
```

Comment:
Instead of requiring a check, let browser.study.validateJSON throw an error on invalid json, which background scripts can catch